src/legacy/core_plugins/kibana/public/visualize/np_ready/editor/editor.js

```diff
@@ -614,6 +629,8 @@ function VisualizeAppController(
       title: savedVis.title,
       type: savedVis.type || stateContainer.getState().vis.type,
     });
+    savedVis.searchSource.setField('query', stateContainer.getState().query);
```

Comment:
extra saving, `filters` and `query` are saved in `$scope.fetch` on every change.

Reply:
in fetch we save it in `vis.data`, when saving we need to put it in the saved vis. before it was one and the same.